pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryBoxing.xml

```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+    xmlns="http://pmd.sourceforge.net/rule-tests"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
+
+    <test-code>
+        <description>pos - new Integer(int)</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    Integer i = new Integer(42);
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>neg - new Integer(String)</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    Integer i = new Integer("42");
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>neg - Integer.valueOf(String)</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    Integer i = Integer.valueOf("42");
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>pos - Integer.valueOf(int)</description>
+        <expected-problems>1</expected-problems>
+        <expected-messages>
+            <message>Unnecessary explicit boxing</message>
+        </expected-messages>
+        <code><![CDATA[
+public class Foo {
+    Integer i = Integer.valueOf(42);
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>pos - new Integer(int)</description>
+        <expected-problems>1</expected-problems>
+        <expected-messages>
+            <message>Unnecessary explicit boxing</message>
+        </expected-messages>
+        <code><![CDATA[
+public class Foo {
+    Integer i = new Integer(42);
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>pos - new Integer(int) with widening</description>
+        <expected-problems>1</expected-problems>
+        <expected-messages>
+            <message>Unnecessary explicit conversion from char to Integer</message>
+        </expected-messages>
+        <code><![CDATA[
+public class Foo {{
+    char c = 0;
+    Integer i = new Integer(c);
+}}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>pos - new Integer(int) assigned to object</description>
+        <expected-problems>1</expected-problems>
+        <expected-messages>
+            <message>Unnecessary explicit boxing</message>
+        </expected-messages>
+        <code><![CDATA[
+public class Foo {{
+    char c = 0;
+    Object obj = new Integer(2);
+}}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>pos - unboxing</description>
+        <expected-problems>3</expected-problems>
+        <expected-messages>
```

Comment:
We should add expected-linenumbers to be clear about the violations....
Not so much relevant here, but in the later test cases, where we have one implicit conversion that should not be reported